### PR TITLE
Restore demo 33

### LINKED ISSUE #406: enable examples-demo33 test project
@matrei

see: https://github.com/grails/grails-testing-support/blob/4.0.x/settings.gradle#L34

> - chore: Temporarily disable `examples-demo33` project

> The example test project does not currently work and blocks
progress. This commit disables it from the build temporarily
until conditions have changed.

### PR DESCRIPTION
**Pull request by jamesfredley**
Requires https://github.com/grails/grails-testing-support/pull/435 to be merged first

Ignores 2 new failing tests with `@PendingFeature` or `@Ignore` 

list of failing tests: https://github.com/grails/grails-testing-support/pull/433/files

issue: https://github.com/grails/grails-testing-support/issues/434

### CODE REVIEW

File: build.gradle [hunk 1]
```diff
@@ -72,7 +72,7 @@ subprojects { project ->
     apply plugin: "groovy"
     if (project.name.startsWith("examples")) {
         apply plugin: "org.grails.grails-web"
-        apply plugin: "org.grails.plugins.views-json"
+        //apply plugin: "org.grails.plugins.views-json"
```
Comment (jamesfredley) [inline build.gradle:75]: Not required for tests here, issue:  https://github.com/grails/grails-views/issues/587

File: examples/demo33/build.gradle [hunk 2]
```diff
@@ -45,17 +45,16 @@ dependencies {
...
-    testRuntimeOnly "net.sourceforge.htmlunit:htmlunit:$htmlunitVersion"
 
     testImplementation project(':grails-web-testing-support')
     testImplementation project(':grails-gorm-testing-support')
 
-    testImplementation "org.seleniumhq.selenium:selenium-remote-driver:$seleniumVersion"
-    testImplementation "org.seleniumhq.selenium:selenium-api:$seleniumVersion"
-    testImplementation "org.seleniumhq.selenium:selenium-support:$seleniumVersion"
-    testImplementation "org.seleniumhq.selenium:selenium-chrome-driver:$seleniumVersion"
+    integrationTestRuntimeOnly "org.grails.plugins:geb"
```
Comment (matrei) [inline examples/demo33/build.gradle]: This has to be `integrationTestImplementation` as the `GebSpec` class is used.

File: build.gradle (hunk 1 above)
Comment (matrei) [reply build.gradle:75]: I think it is used to compile `examples/demo33/grails-app/views/json/index.gson`

File: examples/demo33/src/integration-test/groovy/demo/DependencyInjectionSpec.groovy [hunk 3]
```diff
@@ -1,15 +1,17 @@
...
 class DependencyInjectionSpec extends Specification {
 
     HelperService helperService
 
-    @OnceBefore
+    @Before
+    @RunOnce
     void init() {
         assert helperService != null
     }
```
Comment (matrei) [inline examples/demo33/src/integration-test/groovy/demo/DependencyInjectionSpec.groovy:13]: Isn't this the same as (the Spock way, without the annotations):
```groovy
def setup() {
    assert helperService != null
}
```

**Comment by jamesfredley** [comment]
`-Dgeb.env=chromeHeadless` was required for that last test to complete on github actions

**Review by matrei — APPROVED** [review]

File: examples/demo33/src/integration-test/groovy/demo/DependencyInjectionSpec.groovy (hunk 3 above)
Comment (jamesfredley) [reply examples/demo33/src/integration-test/groovy/demo/DependencyInjectionSpec.groovy:13]: It is.  I should have looked deeper than the deprecated annotation.  Updated.

**Pull request merged**